### DISCUSSION TITLE
The world's slowest chatbot

### DISCUSSION THREAD

**Discussion by JohnZolton** [openai/whisper#491] [Show and tell]
I used whisper to detect questions in "real" time and then passed those questions to a question-answering model and return an answer based on a text file of information. 

It technically worked but overall performance was pretty terrible, about 10 seconds between asking the question and outputting an answer. 
Code [here](https://github.com/JohnZolton/scribe/blob/main/chatbot.py).

**Comment by skol101** [comment]
Is processing done in CPU?

**Reply by JohnZolton** [reply]
If you have a GPU it uses it, if not then CPU

**Reply by skol101** [reply]
Wouldn't that be a reason (CPU) why your bot is so slow to process spoken speech?

**Reply by JohnZolton** [reply]
Whisper processes audio files, not live input. So to get around this it transcribes 5 second chunks and sends that to the QA model. So there's a 0-5 second delay before the question gets detected and processed

**Reply by DonFlymoor** [reply]
If you could detect the end of a sentence it could be much faster to reply
